waspc/data/Generator/templates/react-app/src/crud/_crud.ts

```diff
@@ -0,0 +1,98 @@
+{{={= =}=}}
```

Comment:
For each `crud`, we generate one of these "client API" files. Types are defined right in this file, but I'm not sure if we can do it better and more in the spirit of "full stack type safety"? @sodic

Reply:
I know I said we should avoid importing stuff from the server as much as possible, but you're right. Doing so should it cleaner and less repetitive.

I'd really like to express CRUD as nothing more than a collection of Queries and Actions on the server, and then use the existing frontend mechanisms to transform and extract the correct types (i.e., we'd be treating the crud declaration as nothing more than syntax sugar).

I realize some details prevent us from doing this to the full extent (e.g., different ways of calling crud stuff and regular query stuff on the client). But, aside from such edge cases, does the "crud is just syntax sugar for Queries and Actions" idea seem doable?

Reply:
Bump on this one, I am curious what the answer is!

Reply:
The latest implementation utilizes full stack types and it's in the spirit of what @sodic wrote. We commented on this offline when it was first written, didn't post an update.